### ISSUE TITLE
Error on first load of tide

### ISSUE THREAD

**Issue by wirtsi** [IlanCosman/tide#130]
#### Describe the bug

Starting with fish 3.2.1 or tide 4.2.0 (unclear what caused it), I get 
```
Welcome to fish, the friendly interactive shell
warning: An error occurred while redirecting file '/dev/null'
open: Interrupted system call
```
When I uninstall tide, the message goes away. Also noteworthy is that after the shell starts, special keys (like ctrl-c) don't work but output ^C. After hitting return once, it works.

Another note, this only happens the first time I start my terminal app (kitty in my case). Any subsequent windows don't show the error but still get the swallowed up special chars.

#### Steps to reproduce

* Uprade fish (with brew)
* Ugrade tide (with fisher)
* Observe the error described above

#### Expected behavior
No error message

#### Screenshots

![image](https://user-images.githubusercontent.com/276751/111659996-f4ec2080-880d-11eb-89b1-63eff30bd8e5.png)

#### Environment (please complete the following information)

- Operating System: MacOs 11.2.2
- Output of `tide bug-report`: 

```console
fish, version 3.2.1
tide, version 4.2.0
xterm-kitty
```

**Comment by IlanCosman** [comment]
Thanks for submitting this report 😄 I can't trivially reproduce this, but I'll look into it when I have the time.

**Comment by branchvincent** [comment]
although i'm not seeing any printed warnings, i'm also experiencing issues on the first prompt (same environment):

[![asciicast](https://asciinema.org/a/zWMaGf1Fp2qjus8RzIJSbEWux.svg)](https://asciinema.org/a/zWMaGf1Fp2qjus8RzIJSbEWux)

i've found the following workaround:
```diff
diff --git a/conf.d/_tide_interactive_init.fish b/conf.d/_tide_interactive_init.fish
-function _tide_background_job --on-event fish_prompt --on-variable fish_bind_mode
+function _tide_background_job --on-event fish_prompt
```

**Comment by IlanCosman** [comment]
Yah I noticed that earlier today. I have absolutely not the slightest clue why that change would do that 😩 Maybe I should just go back to the old inelegant method.

**Comment by wirtsi** [comment]
@branchvincent change fixes the issue for me, thanks!

**Comment by martinlindner** [comment]
@branchvincent @IlanCosman:
Had the same issue since yesterday, thanks for the workaround.

Not sure about the details, but I think this started happening right after `brew upgrade`-ing fish 3.2.0 -> fish 3.2.1 for me.

```
> tide bug-report
fish, version 3.2.1
tide, version 4.2.0
screen
```
( screen == iterm2 3.4.4 + tmux 3.1c )

**Comment by IlanCosman** [comment]
https://github.com/fish-shell/fish-shell/issues/7842 just submitted a Fish bug report that should probably solve this.

These things are to be expected when you're on the cutting edge of Fish research 😂

**Comment by IlanCosman** [comment]
Alright, yeah, this willl be fixed in Fish 3.3.0. Hopefully that releases soon.

**Comment by IlanCosman** [comment]
Found a work around! 💪😄

**Comment by paupalou** [comment]
hey @IlanCosman , just fyi, with fish 3.2.2 (released today 7th April) the `</dev/null` part is not needed i think. Works well in my case without the workaround from @branchvincent

**Comment by IlanCosman** [comment]
Yah, they added that fix to 3.2.2 😄
